server/datastore/mysql/software_titles.go

```diff
@@ -306,149 +311,166 @@ func spliceSecondaryOrderBySoftwareTitlesSQL(stmt string, opts fleet.ListOptions
-LEFT JOIN vpp_apps vap ON vap.title_id = st.id AND %s
-LEFT JOIN vpp_apps_teams vat ON vat.adam_id = vap.adam_id AND vat.platform = vap.platform AND %s
-LEFT JOIN software_titles_host_counts sthc ON sthc.software_title_id = st.id AND (%s)
--- placeholder for JOIN on software/software_cve
-%s
--- placeholder for optional extra WHERE filter
-WHERE %s
--- placeholder for filter based on software installed on hosts + software installers
-AND (%s)
-GROUP BY st.id, package_self_service, package_name, package_version, package_platform, package_url, package_install_during_setup, package_storage_id, fleet_maintained_app_id, vpp_app_self_service, vpp_app_adam_id, vpp_app_version, vpp_app_platform, vpp_app_icon_url, vpp_install_during_setup`
-
-	cveJoinType := "LEFT"
-	if opt.VulnerableOnly {
-		cveJoinType = "INNER"
-	}
-
-	countsJoin := "TRUE"
-	softwareInstallersJoinCond := "TRUE"
-	vppAppsJoinCond := "TRUE"
-	vppAppsTeamsJoinCond := "TRUE"
-	includeVPPAppsAndSoftwareInstallers := "TRUE"
-	switch {
-	case opt.TeamID == nil:
-		countsJoin = "sthc.team_id = 0 AND sthc.global_stats = 1"
-		// When opt.TeamID is nil (aka "All teams") we do not include VPP-apps/installers
-		// that are not installed on any host.
-		includeVPPAppsAndSoftwareInstallers = "FALSE"
-	case *opt.TeamID == 0:
-		countsJoin = "sthc.team_id = 0 AND sthc.global_stats = 0"
-		softwareInstallersJoinCond = fmt.Sprintf("si.global_or_team_id = %d", *opt.TeamID)
-		vppAppsTeamsJoinCond = fmt.Sprintf("vat.global_or_team_id = %d", *opt.TeamID)
-	case *opt.TeamID > 0:
-		countsJoin = fmt.Sprintf("sthc.team_id = %d AND sthc.global_stats = 0", *opt.TeamID)
-		softwareInstallersJoinCond = fmt.Sprintf("si.global_or_team_id = %d", *opt.TeamID)
-		vppAppsTeamsJoinCond = fmt.Sprintf("vat.global_or_team_id = %d", *opt.TeamID)
-	}
-
-	if opt.PackagesOnly {
-		vppAppsJoinCond = "FALSE"
-		vppAppsTeamsJoinCond = "FALSE"
-	}
-
-	additionalWhere := "TRUE"
-	match := opt.ListOptions.MatchQuery
-	softwareJoin := ""
-	if match != "" || opt.VulnerableOnly {
-		// if we do a match but not vulnerable only, we want a LEFT JOIN on
-		// software because software installers may not have entries in software
-		// for their software title. If we do want vulnerable only, then we have to
-		// INNER JOIN because a CVE implies a specific software version.
-		softwareJoin = fmt.Sprintf(`
-			%s JOIN software s ON s.title_id = st.id
-			-- placeholder for changing the JOIN type to filter vulnerable software
-			%[1]s JOIN software_cve scve ON s.id = scve.software_id
-		`, cveJoinType)
-	}
-
+	{{if hasTeamID .}}
+		LEFT JOIN software_installers si ON si.title_id = st.id AND si.global_or_team_id = {{teamID .}}
+		LEFT JOIN vpp_apps vap ON vap.title_id = st.id AND {{yesNo .PackagesOnly "FALSE" "TRUE"}}
+		LEFT JOIN vpp_apps_teams vat ON vat.adam_id = vap.adam_id AND vat.platform = vap.platform AND 
+			{{if .PackagesOnly}} FALSE {{else}} vat.global_or_team_id = {{teamID .}}{{end}}
```

Comment:
If `opt.PackagesOnly`, do we technically need to `LEFT JOIN` to `vpp_apps` and `vpp_apps_teams`?  I guess it does make the `SELECT` clause and `GROUP BY` clause a little easier, since the `vat` and `vap` aliases would not exist, and we would have to deal with a additional if statement there.

Does `LEFT JOIN vpp_apps vap ON vap.title_id = st.id AND FALSE` incur any additional resource usage in mysql?

Reply:
AFAIK the query planner should be smart enough to know that `<pred_1> AND FALSE` is always false so I don't expect that to have any performance implications. 

As you well pointed out  the main downside of excluding the JOIN clause and columns if `not opt.PackagesOnly` is that might affect readability by adding an additional if clause (and indentation) and IMHO this query is hard to read already as it is ^^U